main.js

```diff
@@ -0,0 +1,142 @@
+// The app collection
+const AppCollection = Backbone.Collection.extend({
+  // grabs the url thats passed down from the input
+  url() {
+    return this.models[0].attributes.url;
+  },
+  // on the first instance on the add event do the api fetch
+  initialize() {
+    this.on('add', function (model) {
+      model.fetch();
+    });
+  },
+});
+
+// the App Model that references the collection
+const AppModel = Backbone.Model.extend({
+  defaults() {
+    return {
+      videos: new AppCollection(),
+    };
+  },
+});
+
+// the App View that grabs the input and renders the clickable images
+const AppView = Backbone.View.extend({
+  el: 'body',
+
+  // handlebars template
+  template: Handlebars.compile($('#vid-list').html()),
+
+  // event for the click on the input button invokes the add to model
+  events: {
+    'click .search-btn': 'videoSearchAdd',
+  },
+
+  // on instance grab the input id & listene to changes on the videos model and call renderThumbnails
+  initialize() {
+    this.$searchInput = this.$('#search-input');
+    // this.$vidView = $('.vid-container');?
+    this.listenTo(this.model.get('videos'), 'change', this.renderThumbnails);
+  },
+
+  // this function grabs the value of input adds it to the model so it could be grabbed by the url search
+  videoSearchAdd() {
+    const inputValue = this.$searchInput.val();
+
+    const urlInput = `https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=5&q=${inputValue}&type=video&videoEmbeddable=true&key=AIzaSyCn7T_poRhx_PIw7R7PrfqQXYuRJH8RQjw`;
+    this.model.get('videos').add({
```

Comment:
`videos` is a collection -  collections are full of models and all of those models should be of the same kind (in this case, it should _only_ have **video** models. You would never want to do an add like this with the `url`. I think it was just lucky that this worked.